src/ConditionsResolver/Resovler/ConditionalGroup.cs

```diff
@@ -0,0 +1,326 @@
+            }
+            if (foundAnds.Count == 0 && foundOrs.Count == 1)
+            {
+                var splitByOr = ParseOrs(input);
+                Set(splitByOr[0], splitByOr[1].TrimStartExt("OR"), Keyword.OR);
+                return;
+            }
+            if (foundAnds.Count == 0 && foundOrs.Count == 0)
+            {
+                Set(input, string.Empty, Keyword.SINGULAR);
+                return;
+            }
+
+            ParseComplex(input);
+        }
+
+        public void ParseBrackets(string input)
+        {
+            if (string.IsNullOrEmpty(input))
+            {
+                throw new ArgumentNullException(nameof(input));
+            }
+
+            var foundAnds = FindAnds.Matches(input);
+            var foundOrs = FindOrs.Matches(input);
+
+            var foundBrackets = FindBrackets.Match(input);
+            var indexOfClosingBracket = FindCloseBrackets.Match(input).Index;
+            if (!foundBrackets.Success)
+            {
+                throw new InvalidOperationException("Expected Bracket: Bracket not found");
+            }
+            var startingBracketHasBeenTrimmed = indexOfClosingBracket < foundBrackets.Index;
+            // if first index of any ANDs or ORs are before the first bracket then left hand evaluation should be processed
+            if (!startingBracketHasBeenTrimmed && (foundAnds.Any() && foundAnds.First().Index < foundBrackets.Index) || (foundOrs.Any() && foundOrs.First().Index < foundBrackets.Index))
+            {
+                if (!foundOrs.Any() || foundAnds.Any() && foundAnds.First().Index < foundOrs.First().Index)
+                {
+                    var splitByAnds = ParseAnds(input);
+                    var rightAnd = splitByAnds[1].TrimStartExt("AND");
+                    Set(splitByAnds[0], rightAnd, Keyword.AND);
+                }
+                else if (!foundAnds.Any() && foundOrs.Any() || foundOrs.First().Index < foundAnds.First().Index)
+                {
+                    var splitByOrs = ParseOrs(input);
+                    var rightOrs = splitByOrs[1].TrimStartExt("OR");
+                    Set(splitByOrs[0], rightOrs, Keyword.OR);
+                }
+            }
+            else
+            {
+                //handle left hand brackets
+                var bracketedConditionalGroup = input.Substring(0, indexOfClosingBracket);
+                LeftGroup = ConditionalGroup.Create(bracketedConditionalGroup, GroupedLogical);
+                var rightSideConditionalGroup = input.Substring(indexOfClosingBracket + 1, input.Length - indexOfClosingBracket - 1);
+                rightSideConditionalGroup = rightSideConditionalGroup.TrimStartExt("AND").TrimStartExt("OR");
+                RightGroup = ConditionalGroup.Create(rightSideConditionalGroup, GroupedLogical);
+            }
+        }
+
+        private void ParseComplex(string input)
+        {
+            var foundBrackets = FindBrackets.Matches(input);
```

Comment:
May be a good idea to always check for null/whitespace parameters in case anotehr engineer tries to use it incorrectly.